src/workerd/api/r2-bucket.c++

```diff
@@ -819,17 +922,24 @@ R2Bucket::UnwrappedConditional::UnwrappedConditional(jsg::Lock& js, Headers& h)
   }
 }
 
+kj::Array<R2Bucket::Etag> buildSingleStrongEtagArray (kj::String etagValue) {
+  struct R2Bucket::StrongEtag etag = {.value = kj::str(etagValue)};
+  kj::ArrayBuilder<R2Bucket::Etag> etagArrayBuilder = kj::heapArrayBuilder<R2Bucket::Etag>(1);
+  etagArrayBuilder.add(kj::mv(etag));
+  return etagArrayBuilder.finish();
+}
+
 R2Bucket::UnwrappedConditional::UnwrappedConditional(const Conditional& c)
   : secondsGranularity(c.secondsGranularity.orDefault(false)) {
   KJ_IF_MAYBE(e, c.etagMatches) {
     JSG_REQUIRE(!isQuotedEtag(e->value), TypeError,
       "Conditional ETag should not be wrapped in quotes (", e->value, ").");
-    etagMatches = kj::str(e->value);
+    etagMatches = buildSingleStrongEtagArray(kj::str(e->value));
```

Comment:
nit: can you pass a pointer here and get rid of one of the copies?

Reply:
That's a good suggestion :) I've done that now.